taskvine/src/worker/vine_worker.c

```diff
@@ -208,6 +211,66 @@ __attribute__((format(printf, 2, 3))) void send_message(struct link *l, const ch
 	va_end(va);
 }
 
+/* Send messages from list of asychronous messages available*/
+
+void deliver_async_messages(struct link *l)
+{
+	int bytes_in_buffer = get_link_buffer_bytes(l);
+	int bytes_available = VINE_MAX_BUFFER_SIZE - bytes_in_buffer;
+	int messages = list_size(pending_async_messages);
+	int visited;
+	char *message;
+
+	for (visited = 0; visited < messages; visited++) {
```

Comment:
Hmm, I don't think you want to mix up the order of the messages.
Just peek the head and send it if there is space available.
If there isn't room for the first item in the queue, stop sending altogether.